sgf-viewer/src/components/GameList.svelte

```diff
@@ -0,0 +1,138 @@
+<script lang="ts">
+    import GoBoard from "./subcomponents/GoBoard.svelte";
+    export let basePath: string;
+
+    let games: Array<Array<string>> = [];
+    let selectedRow: number = 0;
+    let sgfIndex: number = 0;
+    let sgfPath: string = "";
+    const tableColumns = {
+        victim_color: "Victim Color",
+        win_color: "Win color",
+        adv_win: "Adversary Win",
+        adv_minus_victim_score: "Point difference",
+        num_moves: "Game length",
+        sgf_path: "SGF Path", // Should also include line number
+    };
+    function indexToSgfPath(index: number) {
+        let keyIndex = Object.keys(tableColumns).indexOf("sgf_path");
+        let fileName = games[index][keyIndex].split("/").slice(-1)[0];
+        return `/sgfs/${basePath}/${fileName}`;
+    }
+    $: if (games.length > 0) sgfPath = indexToSgfPath(sgfIndex);
+    function processData(text: string) {
+        let allTextLines = text.trim().split(/\r\n|\n/);
+        let headers = allTextLines[0].split(",");
+        // Get the indices from headers of the tableColumns keys
+        let tableColumnIndices: Array<number> = Object.keys(tableColumns).map(
+            (key) => headers.indexOf(key)
+        );
+
+        // for (var i = 1; i < allTextLines.length; i++) {
+        for (var i = 1; i < Math.min(allTextLines.length, 10); i++) {
+            var data = allTextLines[i].split(",");
+            // Push the values of the tableColumns keys into the games array
+            games.push(tableColumnIndices.map((index) => data[index]));
+        }
+        games = games; // trigger Svelte reactivity. See https://svelte.dev/tutorial/updating-arrays-and-objects
+    }
+
+    function handleRowClick(rowIndex: number) {
+        sgfIndex = rowIndex;
+    }
+    fetch(`/sgfs/${basePath}/game_infos.csv`)
+        .then((r) => r.text())
+        .then((text) => processData(text));
+</script>
+
+<div>
+    <table>
+        <!-- Loop through headers and games and put them in table -->
+        <tr>
+            {#each Object.values(tableColumns).slice(0, -1) as header}
```

Comment:
Maybe add a comment that `.slice(0, -1)` is removing the "SGF Path" column here.